internal/common/allow.go

```diff
@@ -0,0 +1,81 @@
+package common
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+// the ALLOW_FILE file must conform to the following schema:
+// {
+// 	"default":{
+//		"rhel-84": true,
+//		"rhel-85": true,
+//		"rhel-86": true,
+//		"rhel-90": true,
+//		"centos-8": true,
+//		"centos-9": true,
+// 	},
+// 	"000000":{
+//		"rhel-84": true,
+//		"rhel-85": true,
+//		"rhel-86": true,
+//		"rhel-90": true,
+//		"centos-8": true,
+//		"centos-9": true,
+// 	},
+// }
+
+// Fallback defaults used if an allowFile is not specified
+var defaultAllowList = map[string]bool{
+	"rhel-84":  true,
+	"rhel-85":  true,
+	"rhel-86":  true,
+	"rhel-90":  true,
+	"centos-8": true,
+	"centos-9": true,
+}
+
+func isAllowed(distribution string, allowList map[string]bool) (bool, error) {
+	if allow, ok := allowList[distribution]; ok {
+		return allow, nil
+	} else {
+		return false, fmt.Errorf("%s is not an allowable distribution", distribution)
+	}
+}
+
+// Returns true if the distribution requested is allowed to be built by the requesting account's organization.
+// The orgId is cross referenced against an allow list file which is pointed to by the ALLOW_FILE environment
+// variable. In the case that the allowFile parameter is an empty string (as would occur if the ALLOW_FILE
+// environment variable was not set) then a fallback default allow list is used instead.
+func CheckAllow(orgId string, distribution string, allowFile string) (bool, error) {
+	if allowFile == "" {
+		return isAllowed(distribution, defaultAllowList)
+	}
+
+	var allowList map[string]map[string]bool
+
+	jsonFile, err := os.Open(filepath.Clean(allowFile))
+	if _, error := err.(*os.PathError); error {
```

Comment:
The case when `err` isn't `*os.PathError` also needs to be handled.

Reply:
In fact os.Open can *only* return an *os.PathError, so I changed this to a more general `if err != nil` block with the same error message.